libraries/aws/shadow/src/shadow.c

```diff
@@ -0,0 +1,550 @@
+
+/**
+ * @brief The length of "/shadow/get/accepted".
+ */
+
+/* MISRA Rule 5.4 flags the following macro's name as ambiguous from the
+ * one without _ACCEPTED suffix. This rule is suppressed for naming consistency with
+ * other Shadow header field and value string and length macros in this file.*/
+/* coverity[misra_c_2012_rule_5_4_violation] */
+#define SHADOW_TOPIC_OPERATION_LENGTH_GET_ACCEPTED                  ( SHADOW_TOPIC_OPERATION_LENGTH_GET + SHADOW_TOPIC_SUFFIX_LENGTH_ACCEPTED )
+
+/**
+ * @brief The length of "/shadow/get/rejected".
+ */
+
+/* MISRA Rule 5.4 flags the following macro's name as ambiguous from the
+ * one without _REJECTED suffix. This rule is suppressed for naming consistency with
+ * other Shadow header field and value string and length macros in this file.*/
+/* coverity[misra_c_2012_rule_5_4_violation] */
+#define SHADOW_TOPIC_OPERATION_LENGTH_GET_REJECTED                  ( SHADOW_TOPIC_OPERATION_LENGTH_GET + SHADOW_TOPIC_SUFFIX_LENGTH_REJECTED )
+
+/**
+ * @brief The length of "/shadow/get/accepted".
+ */
+
+/* MISRA Rule 5.4 flags the following macro's name as ambiguous from the
+ * one without _ACCEPTED suffix. This rule is suppressed for naming consistency with
+ * other Shadow header field and value string and length macros in this file.*/
+/* coverity[misra_c_2012_rule_5_4_violation] */
+#define SHADOW_TOPIC_OPERATION_LENGTH_DELETE_ACCEPTED               ( SHADOW_TOPIC_OPERATION_LENGTH_DELETE + SHADOW_TOPIC_SUFFIX_LENGTH_ACCEPTED )
+
+/**
+ * @brief The length of "/shadow/delete/rejected".
+ */
+
+/* MISRA Rule 5.4 flags the following macro's name as ambiguous from the
+ * one without _REJECTED suffix. This rule is suppressed for naming consistency with
+ * other Shadow header field and value string and length macros in this file.*/
+/* coverity[misra_c_2012_rule_5_4_violation] */
+#define SHADOW_TOPIC_OPERATION_LENGTH_DELETE_REJECTED               ( SHADOW_TOPIC_OPERATION_LENGTH_DELETE + SHADOW_TOPIC_SUFFIX_LENGTH_REJECTED )
+
+/**
+ * @brief Determine if the string contains the substring.
+ *
+ * @param[in] pString Pointer to the string.
+ * @param[in] stringLength Length of pString.
+ * @param[in] pSubString Pointer to the substring.
+ * @param[in] subStringLength Length of pSubString.
+ *
+ * @return Return SHADOW_STATUS_SUCCESS if it contains;
+ *         return SHADOW_STATUS_FAIL if not.
+ */
+static ShadowStatus_t containsSubString( const char * pString,
+                                         uint16_t stringLength,
+                                         const char * pSubString,
+                                         uint16_t subStringLength );
+/**
+ * @brief Extract the Thing Name from a string.
+ *
+ * @param[in] pString Pointer to the string.
+ * @param[in] stringLength Length of pString.
+ * @param[out] pThingName Points to the 1st character of Thing Name inside of the topic string.
```

Comment:
I don't think this parameter is necessary. Since it's always set to `pString`, the calling function can just set the pointer directly whenever this function returns success.

Reply:
same above

Reply:
fixed it in commit 8da68c5

by removing pThingName for extractThingName, and assign it in Shadow_MatchTopic

resolving this comment

feel free to let me know if it needs further changes, thank you